types/jquery/jquery-tests.ts

```diff
@@ -6085,11 +6085,21 @@ function JQuery() {
             $('p').replaceWith([new HTMLElement()]);
 
             // $ExpectType JQuery<HTMLElement>
-            $('p').replaceWith(function() {
+            $('p').replaceWith(document.createTextNode('bar'));
+
+            // $ExpectType JQuery<HTMLElement>
+            $('p').replaceWith(document.createComment('bar'));
+
+            // $ExpectType JQuery<HTMLElement>
+            $('p').replaceWith(function(index, oldhtml) {
                 // $ExpectType HTMLElement
                 this;
+                // $ExpectType number
+                index;
+                // $ExpectType string
+                oldhtml;
 
-                return this;
+                return undefined! as JQuery.htmlString | JQuery<JQuery.Node> | JQuery.TypeOrArray<Element> | JQuery.Node;
```

Comment:
Use `as` in test.ts is a bad smell. Why not returning `this` in this method and declare returning type as `JQuery`?

Reply:
This is intended to concisely test that `.replaceWith()` accepts a callback that returns all possible return types (or any combination of them). Returning only `this` would only test that `HTMLElement` is a valid return type.

Reply:
If we want to test all possible returning cases, they should be list here one by one. 

```
{ ... return this; }
{ ... return $('div'); }
{ ... return $('<div />'); }
```

Reply:
This is also intended to test scenarios where a user writes code that results in a union of the potential return types (e.g. `JQuery.htmlString | Element`). Testing all possible combinations results in a lot of tests and I'm not sure there's even a real gain to doing so.

I use the type assertion like that because what I really need is to test the return type. The code that gets me there shouldn't matter.